spec/controllers/concerns/collaborator_processing_spec.rb

```diff
@@ -114,4 +114,39 @@ class FakesController < ApplicationController
       end
     end
   end
+
+  context 'removing users' do
+    let!(:hank) { create(:user, first_name: 'Hank') }
+    let!(:hanky) { create(:user, first_name: 'Hanky') }
+
+    let!(:collaborator) { create(:cookbook_collaborator, resourceable: cookbook, user: hank) }
+
+    it 'deletes a collaborator if the signed in user is the resource owner' do
+      sign_in fanny
+
+      expect do
+        subject.remove_collaborator(collaborator)
+      end.to change { Collaborator.count }.by(-1)
+
+      expect(response).to be_success
+    end
+
+    it 'deletes a collaborator if the signed in user is a collaborator on this resource' do
```

Comment:
The description of this spec is ambiguous to me. Is this scenario "a user can remove their own collaboration access" or "a collaborator can remove another collaborator's access"? Code implies the former, but I'm not sure of the intention.